### DISCUSSION TITLE
Filter in widget by metadata

### DISCUSSION THREAD

**Discussion by karimj** [pretix/pretix#2509] [Q&A] [answered]
Hi!

I have multiple event series with lots of events in each of the series. I would like to use the widget to display a calendar with only a subset of these event series. I thought I could manage that by adding a meta-data property to certain event series (e.g. type=xyz) and then use the filter property of the widget (filter="attr[type]=xyz"). However, I can't manage to get that to work. Maybe I misunderstood the feature. Is it possible to achieve what I want?

Best
Karim

**Comment by raphaelm** [comment, marked as answer]
Yes, this should be possible. Please post a full snippet so we can try to guess what the problem is :)

**Reply by karimj** [reply]
This is my widget Code:
`<pretix-widget event="https://tickets.happylab.at/happylab.wien/" style="calendar" filter="attr[Typ]=kiget"></pretix-widget>`

The result just shows all event series regardless, but there is only one with the Typ metadata property set.

**Reply by wiffbi** [reply]
I tested it with an event series and marked only one date with meta-data. It works as expected with the widget-filter. One thing to note though: make sure in the event series’ settings (not the settings of a single date/subevent) the meta data is not set. Inside a date/subevent you cannot overwrite the metadata of the containing series with an empty value.

**Reply by karimj** [reply]
Thanks for testing this, maybe I have the wrong settings?
I select the event series, then go to Settings > General > Item Metadata. There I can create a Property and Value
But where I can set them for the individual dates?

**Reply by wiffbi** [reply]
Ah, that is the wrong place – this is metadata for items a.k.a. „Products“ (we probably should update the wording in that description).
What you want is metadata on the event. For this to work, you need to create a metadata-property on organizer level. You go to your organizer, then Settings > Event metadata. There you add which metadata is available.
After that you select the event > Settings > General > Tab „Basics“ and scroll down to the end. There is a section „Meta data“. This is also available for the whole event-series.
If you want to tag only one date in an event-series, then do not set the metadata on the event/event-series as described above, but do the following: inside the event-series under „Dates“, select the date to edit it. At the bottom of the fist section „General information“ you will find the metadata settings.

**Reply by karimj** [reply]
Thank you very much! This was exactly my problem.

One additional short question: Is it possible to use other logical operators (!=) and combine multiple attributes (AND, OR)?

**Reply by wiffbi** [reply]
No, currently not. Only AND is possible by adding another condition for e.g. a metadata „Art“ like so:

`filter="attr[Typ]=kagit&attr[Art]=ticket"`

**Reply by karimj** [reply]
Okay, thank!
